services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,944 @@
+        flowHelper.deleteFlow(flow.id)
+        database.resetCosts()
+
+        where:
+        flowDescription | bandwidth
+        "metered"       | 1000
+        "unmetered"     | 0
+    }
+
+    @Unroll
+    def "System reroutes #flowDescription flow to more preferable path and ignore protected path in case reroute is intentional"() {
+        given: "A flow with alternate paths available"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> allPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.unique { it.sort() }.find { Switch src, Switch dst ->
+            allPaths = database.getPaths(src.dpId, dst.dpId)*.path
+            allPaths.size() > 2
+        } ?: assumeTrue("No suiting switches found", false)
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        flowPathInfo.protectedPath
+
+        def currentPath = pathHelper.convert(flowPathInfo)
+        def currentProtectedPath = pathHelper.convert(flowPathInfo.protectedPath)
+        currentPath != currentProtectedPath
+
+        when: "Make the current and protected path less preferable than alternatives"
+        def alternativePaths = allPaths.findAll { it != currentPath && it != currentProtectedPath }
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, currentPath) }
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, currentProtectedPath) }
+
+        and: "Init reroute"
+        def rerouteResponse = northbound.rerouteFlow(flow.id)
+
+        then: "Flow is rerouted"
+        rerouteResponse.rerouted
+
+        and: "Path is not changed to protected path"
+        def flowPathInfoAfterRerouting = northbound.getFlowPath(flow.id)
+        def newCurrentPath = pathHelper.convert(flowPathInfoAfterRerouting)
+        newCurrentPath != currentPath
+        newCurrentPath != currentProtectedPath
+        def newCurrentProtectedPath = pathHelper.convert(flowPathInfoAfterRerouting.protectedPath)
+        newCurrentProtectedPath != currentPath
+        newCurrentProtectedPath != currentProtectedPath
+        Wrappers.wait(WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Cleanup: revert system to original state"
+        flowHelper.deleteFlow(flow.id)
+        database.resetCosts()
+
+        where:
+        flowDescription | bandwidth
+        "metered"       | 1000
+        "unmetered"     | 0
+    }
+
+    @Unroll
+    def "System reroutes #flowDescription flow to protected path and ignore more preferable path in case reroute is automatical"() {
+        given: "A flow with alternate paths available"
+        def switches = topology.getActiveSwitches()
```

Comment:
Can we use getNotNeighboringSwitchPair(2) here?

Reply:
sure